SilKit/source/core/vasio/VAsioPeer.hpp

```diff
@@ -121,9 +122,8 @@ class VAsioPeer
 
     // receiving
     std::atomic<uint32_t> _currentMsgSize{0u};
-    std::vector<uint8_t> _msgBuffer;
-    size_t _wPos{0};
-    MutableBuffer _currentReceivingBuffer;
+    RingBuffer<uint8_t> _msgBuffer;
```

Comment:
Shouldn't we make this a configurable option? Pros: we could easily compare performance, have users switch back, if we suspect trouble. cons: we'd have both code paths lying around.

Reply:
I think it would be nice to have only one code path and rely on the ring buffer tests. Do you think the tests cover all important corner cases?

Reply:
Right, having a single code path and good test coverage is enough.